Automattic/wp-calypso
# Add the authorization Step for the Application Passwords

Related to #

## Proposed Changes

Adds the new Authorization Step to allow us to capture the Application Password authorization. This step gives the user the possibility to authorize our application so we can use the credentials for the automate checks and tasks.

I also added the possibility of adding `notices` to the Step. I updated the StepContainer to accept a new prop: `notice`. And it is displayed before the Formatted Header.

### Default view

![Captura de Tela 2024-12-03 às 20 23 56](https://github.com/user-attachments/assets/0eef25cf-81fc-4eab-b323-509e6220ee0c)

### Rejected the Authorization

![Captura de Tela 2024-12-03 às 20 23 43](https://github.com/user-attachments/assets/868ecbb1-2ec8-411e-885c-ca04cfc6f1f8)

### Error trying to store the Authorization credentials

![Captura de Tela 2024-12-03 às 20 23 23](https://github.com/user-attachments/assets/3a7d1781-909a-4286-a03b-164efde2df3d)


## Why are these changes being made?


* This is an attempt to improve the low success rates of users credentials. Most of the credentials that the users inputs are not valid and we can proceed with many automated processes that would save a great deal of time for the HEs. More context on:  paYKcK-5B6-p2

## Testing Instructions



_Visual inspection should be enough, as I added some tests to cover the cases._

To test it manually follow the steps:

* Use calypso live to avoid the SSL error when trying to authenticate.
* Go through the migration process until you reach the Credentials step.
* Then update the URL to point to the `/setup/site-migration/site-migration-application-password-authorization` path. Keep the current parameters, and add the `authorizationUrl`, which could be a valid authorization URL for the Application Passwords: `https://example.com/wp-admin/authorize-application.php&app_id=1083d914-d9f8-47ec-b49b-55d206104730&app_name=Test`
* You should see the default view, inviting the user to Authorize. Click on the "Authorize" button.
* You should be redirected to your test site.
* Reject the request, and you should be redirected to the Authorization step with an alert.
* Click again on the "Authorize" button, and this time approve the request.
* You should be redirected to the Authorization step and should see a loading state.
* After the loading state, you'll probably see the Authorization step with an error message, because the endpoint that stores the Password will fail because you don't have the ticket.
* So far this should be enough to test.

## Pre-merge Checklist
